source/JsMaterialX/JsMaterialXCore/JsDocument.cpp

```diff
@@ -17,14 +18,16 @@ extern "C"
         ems::class_<mx::Document, ems::base<mx::GraphElement>>("Document")
             .smart_ptr_constructor("Document", &std::make_shared<mx::Document, mx::ElementPtr, const std::string &>)
             .smart_ptr<std::shared_ptr<const mx::Document>>("Document")
+             // At the moment only the Document type is used. Once more types are added this binding needs to be adapted accordingly.
```

Comment:
@muenstc What does this comment mean?

Reply:
Currently in the c++ code the only call to the static class function `createDocument` is using the `Document` type.
That is why there is only this one binding.
If we followed the usual pattern of the Naming for typed Js Bindings it would actually become `createDocumentDocument` which is rather weird.
The comment is just to indicate that this would need to be revised if usages of `createDocument` with other types pop up.

Reply:
I see. It's probably fine to only expose this for Document now, even though you can use it with other types in the C++ API, of course. I'd suggest to still call this `createDocument` for the Document type later, and only append the other types to the name.